pkg/apis/controller/v1beta1/appwrapper.go

```diff
@@ -83,19 +100,27 @@ type AppWrapperResource struct {
 	MinAvailable *int32 `json:"minavailable,omitempty" protobuf:"bytes,3,opt,name=minavailable"`
 
 	// The number of allocated replicas from this resource type
+	// +optional
 	AllocatedReplicas int32 `json:"allocatedreplicas"`
 
-	// The priority of this resource
-	Priority float64 `json:"priority"`
+	// +kubebuilder:validation:Type=number
+	// +kubebuilder:validation:Format=int
+	// +optional
+	Priority float64 `json:"priority,omitempty"`
 
 	// The increasing rate of priority value for this resource
+	// +optional
+	// +kubebuilder:validation:Type=number
+	// +kubebuilder:validation:Format=float
 	PrioritySlope float64 `json:"priorityslope"`
 
 	//The type of the resource (is the resource a Pod, a ReplicaSet, a ... ?)
+	// +optional
```

Comment:
This is required. Will need to reflect in schema

Reply:
Done